### PR TITLE
Session management feature 

### PR DESCRIPTION
**Pull request by salman90**
## Purpose

* This PR introduces a new feature to the sample where the SPA application will proactively check the id token expiry time. If the id token has expired, the user will be to reauthenticate silently. 

## Does this introduce a breaking change?

```
[ x ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/Azure-Samples/ms-identity-javascript-react-tutorial.git
cd  cd  .\2-Authorization-I\1-call-graph\SPA\
git checkout session-managment
npm install 
npm start
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                instance
+                    .ssoSilent({
+                        ...loginRequest,
+                        account: account,
+                    })
+                    .catch((error) => {
+                        if (error instanceof InteractionRequiredAuthError) {
+                            instance.loginRedirect({
+                                ...loginRequest,
+                                prompt: 'login',
```
Comment (derisen) [inline 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx:26]: do we actually need login prompt here?

File: 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { InteractionRequiredAuthError, InteractionStatus } from '@azure/msal-browser';
+import { loginRequest } from '../authConfig';
+import moment from 'moment';
+
+export const SessionManager = ({ children }) => {
+    const { instance, inProgress } = useMsal();
+    const account = instance.getActiveAccount();
+    useEffect(() => {
+        if (account && inProgress === InteractionStatus.None) {
+            let endDate = moment(new Date(account.idTokenClaims.exp * 1000));
```
Comment (tnorling) [inline 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx:12]:
```suggestion
            let expiresAt = moment(new Date(account.idTokenClaims.exp * 1000));
```

File: 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            let startData = moment(new Date());
```
Comment (tnorling) [inline 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx:13]:
```suggestion
            let currentTime = moment(new Date());
```

File: 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx (hunk 1 above)
Comment (tnorling) [reply 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx:26]: +1 we should let eSTS decide what prompt is needed, if any

**Review by tnorling — APPROVED** [review]

**Review by derisen — APPROVED** [review]

File: 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx (hunk 1 above)
Comment (salman90) [reply 2-Authorization-I/1-call-graph/SPA/src/components/SessionManager.jsx]: I removed the login prompt. Thanks for the feedback!

**Pull request merged**